server/paymentProviders/stripe/webhook.ts

```diff
@@ -47,11 +53,15 @@ const paymentIntentSucceeded = async (event: Stripe.Response<Stripe.Event>) => {
   const transaction = await createChargeTransactions(charge, { order });
 
   await order.update({
-    status: OrderStatuses.PAID,
+    status: !order.SubscriptionId ? OrderStatuses.PAID : undefined,
     processedAt: new Date(),
     data: { ...order.data, paymentIntent },
   });
 
+  if (order.interval && !order.SubscriptionId) {
+    await createSubscription(order);
+  }
```

Comment:
This should be in the `processRecurringOrder` logic.

Reply:
The processRecurringOrder is only called as part of the recurring-contributions cron job, it creates and confirms a payment intent off session. 

Therefore, setting up the subscription should come before it, doing it on this webhook ensures us that the first payment was successful and there is a reasonable expectation for the next charges to go through off session.